src/biosHandler.cpp

```diff
@@ -8,7 +8,11 @@
 
 namespace vpd
 {
-void BiosHandler::checkAndListenPldmService()
+// Template declaration to define APIs.
+template class BiosHandler<IbmBiosHandler>;
+
+template <typename T>
+void BiosHandler<T>::checkAndListenPldmService()
```

Comment:
can this API throws an exception is any scenario ?

Reply:
Going by the current implementation. It should not.